src/functions.php

```diff
@@ -42,6 +43,39 @@
 use function reset;
 use function substr;
 
+/**
+ * Check whether all servers support executing a write stage on a secondary.
+ *
+ * @internal
+ * @see https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#aggregation-pipelines-with-write-stages
+ */
+function all_servers_support_write_stage_on_secondary(Manager $manager): bool
+{
+    static $wireVersionForWriteStageOnSecondary = 13;
+
+    $servers = $manager->getServers();
+
+    /* If no servers are available, assume that SDAM has not been initialized
+     * and make a single server selection attempt to do so now. We do not
+     * actually care about selecting a server so any error can be ignored. */
+    if (empty($servers)) {
+        try {
+            $manager->selectServer(new ReadPreference(ReadPreference::NEAREST));
+        } catch (DriverRuntimeException $e) {
```

Comment:
A server selection failure typically comes up as a ConnectionTimeoutException, but the extenion's RuntimeException is the base class and also suitable to work with here.